server/scriptController.js

```diff
@@ -19,11 +21,110 @@ const scriptController = {
       })
       .catch((error) => {
         return next({
-          log: `error: ${error} occured when getting script titles`,
-          message: 'error in getScriptTitles in scriptController' ,
+          log: `error: ${error} occurred when getting script titles`,
+          message: 'error in getScriptTitles in scriptController',
         });
       });
   },
+  saveScript: (req, res, next) => {
+    const MAX_FILESIZE_BYTES = 50 * 1024 * 1024; //50MB. If updating, change constant in Upload.jsx too.
+
+    // configure multer storage to upload files to the path from the .env file
+    const storage = multer.diskStorage({
+      destination: function (req, file, cb) {
+        cb(null, process.env.UPLOADPATH);
+      },
+      filename: function (req, file, cb) {
+        cb(null, Date.now() + file.originalname);
+      },
+    });
+
+    function fileFilter(req, file, cb) {
+      const name = file.originalname;
+      const regex = /^[a-zA-Z0-9](?:[a-zA-Z0-9 ._-]*[a-zA-Z0-9])?\.[a-zA-Z0-9_-]+$/;
```

Comment:
Source: [](https://stackoverflow.com/questions/11794144/regular-expression-for-valid-filename/41446229#41446229)

This seemed like the most reputable answer.

Reply:
I'd suggest removing the file extension component of the regex since that's not important for our use. Also, could you inline the regex variable?